File: Trimestre 6/Proyecto final/API REST/src/models/categoria.model.js
// models/categoria.model.js
import { Model, DataTypes } from "sequelize";
import sequelize from "../config/connect.db.js";

class Categoria extends Model {}

Categoria.init(
  {
    idCategoria: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
      field: "idCategoria",
    },
    Nombre_Categoria: {
      type: DataTypes.STRING(60),
      allowNull: false,
      field: "Nombre_Categoria",
    },
    Descripcion: {
      type: DataTypes.STRING(200),
      allowNull: false,
      field: "Descripcion",
    },
    Estado: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
      field: "Estado",
    },
  },
  {
    sequelize,
    modelName: "Categoria",
    tableName: "categorias", // como la tienes en la BD
    timestamps: false,
  }
);

export default Categoria;

File: Trimestre 6/FlutterProjects/Flutter/frontend/lib/models/categoria_model.dart
class Categoria {
  int idCategoria;
  String nombreCategoria;
  String descripcion;

  Categoria({
    required this.idCategoria,
    required this.nombreCategoria,
    required this.descripcion,
  });

  factory Categoria.fromJson(Map<String, dynamic> json) {
    return Categoria(
      idCategoria: json['idCategoria'],
      nombreCategoria: json['Nombre_Categoria'] ?? json['nombreCategoria'],
      descripcion: json['Descripcion'] ?? json['descripcion'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "Nombre_Categoria": nombreCategoria,
      "Descripcion": descripcion,
    };
  }
}
